src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -147,8 +147,8 @@ public StorageHelper(Context ctx) throws NoSuchAlgorithmException, NoSuchPadding
     final private void loadSecretKeyForAPI() throws NoSuchAlgorithmException,
             InvalidKeySpecException {
         // Loading key only once for performance. If API is upgraded, it will
-        // restart the device anyway. It will load the correct key for new API.
-        if (sKey != null && sMacKey != null)
+        // restart the device anyway. It will load the correct key for new API. 
+        if (mKey != null && mMacKey != null)
```

Comment:
In case of company portal, because authenticator process executes within CP process, the removal of static is not sufficient. You need to remove the entire if statement.